tests/std/tests/GH_002030_asan_annotate_string/test.cpp

```diff
@@ -1917,17 +1919,32 @@ void test_gh_3955() {
     assert(s == t);
 }
 
-int main() {
-    run_allocator_matrix<char>();
+void test_gh_5251() {
+    // GH-5251 <string>: ASan annotations do not prevent writing to allocated
+    // but uninitialized basic_string memory
+    std::basic_string<char> myString;
+    myString.reserve(100);
+    char* data = &myString[0];

```

Comment:
Nitpick: With `<string>` and `using namespace std;`, this shadows the [`std::data` function](https://en.cppreference.com/w/cpp/iterator/data). Let's rename this slightly.